controllers/space/nstemplatetier_spaces_mapper_test.go

```diff
@@ -0,0 +1,134 @@
+package space_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	tierutil "github.com/codeready-toolchain/host-operator/controllers/nstemplatetier/util"
+	"github.com/codeready-toolchain/host-operator/controllers/space"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	tiertest "github.com/codeready-toolchain/host-operator/test/nstemplatetier"
+	spacetest "github.com/codeready-toolchain/host-operator/test/space"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestMapNSTemplateTierToSpaces(t *testing.T) {
+
+	// given
+	logf.SetLogger(zap.New(zap.UseDevMode(true)))
+	err := apis.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+
+	nsTmplTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+	otherSpace1 := spacetest.NewSpace("other-space-1", spacetest.WithTierNameAndHashLabelFor(nsTmplTier))
+	otherSpace2 := spacetest.NewSpace("other-space-2", spacetest.WithTierNameAndHashLabelFor(nsTmplTier))
+	otherSpace3 := spacetest.NewSpace("other-space-3", spacetest.WithTierNameAndHashLabelFor(nsTmplTier))
```

Comment:
as I mentioned in the comment: https://github.com/codeready-toolchain/host-operator/pull/597#discussion_r812918529
I was talking about adding a Space for a different tier. Something like this:
```suggestion
	otherSpace2 := spacetest.NewSpace("other-tier-space",
		spacetest.WithLabel(tierutil.TemplateTierHashLabelKey("other-tier"), "123abc"))
	otherSpace3 := spacetest.NewSpace("no-label-space")
```